DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/AdminFieldsController.cs

```diff
@@ -28,22 +28,31 @@ public class AdminFieldsController : Controller
         private static readonly DateTimeOffset CookieExpiry = DateTimeOffset.UtcNow.AddDays(7);
         private readonly ICourseAdminFieldsDataService courseAdminFieldsDataService;
         private readonly ICourseAdminFieldsService courseAdminFieldsService;
+        private readonly ICourseService courseService;
 
         public AdminFieldsController(
             ICourseAdminFieldsService courseAdminFieldsService,
-            ICourseAdminFieldsDataService courseAdminFieldsDataService
+            ICourseAdminFieldsDataService courseAdminFieldsDataService,
+            ICourseService courseService
         )
         {
             this.courseAdminFieldsService = courseAdminFieldsService;
             this.courseAdminFieldsDataService = courseAdminFieldsDataService;
+            this.courseService = courseService;
         }
 
         [HttpGet]
         [Route("{customisationId}/AdminFields")]
         public IActionResult Index(int customisationId)
         {
+            if (!VerifyAdminUserCanAccessCourse(customisationId))
+            {
+                return NotFound();
+            }
+
             var centreId = User.GetCentreId();
             var categoryId = User.GetAdminCategoryId()!;
+
             var courseAdminFields = courseAdminFieldsService.GetCustomPromptsForCourse(

```

Comment:
Do we need the categoryId here - looks like it's not actually used in the DB query 🤔

Reply:
Changed the method to return a boolean if the course exists or not :)